internal/services/mssqlmanagedinstance/mssql_managed_database_resource.go

```diff
@@ -319,6 +335,10 @@ func (r MsSqlManagedDatabaseResource) Read() sdk.ResourceFunc {
 				model.PointInTimeRestore = flattenManagedDatabasePointInTimeRestore(v)
 			}
 
+			if model.Tags != nil {
+				state.Tags = model.Tags
```

Comment:
You could get rid of this block and just do `state.Tags = pointer.From(model.Tags)`